src/enzyme_ad/jax/Implementations/WhileLoopInfo.h

```diff
@@ -0,0 +1,47 @@
+//===- WhileLoopInfo.h - While Op range analysis --------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "stablehlo/dialect/StablehloOps.h"
+
+using namespace mlir;
+using namespace mlir::stablehlo;
+
+namespace mlir {
+
+namespace enzyme {
+
+struct WhileLoopInfo {
+  WhileOp op;
+
+  mlir::Value start;   // garanteed to dominate the while op
+  mlir::Value limit;   // not garanteed to dominate the while op
+  mlir::Value step; // not garanteed to dominate the while op
+
+  WhileLoopInfo(WhileOp op_) : op(op_) {}
+
+  LogicalResult computeInfo();
+
+  bool isValid() { return start && limit && step; }
+  bool isConstant() {
+    return getConstantStep().has_value() &&
+           getConstantStart().has_value() &&
+           getConstantLimit().has_value();
+  }
+
+  std::optional<int64_t> getConstantStep();
+  std::optional<int64_t> getConstantStart();
+  std::optional<int64_t> getConstantLimit();
+
+  /// Needs to be constant
```

Comment:
Yeah here for the get number of induction variables, we can take a builder and compute the number of iterations